packages/block-library/src/quote/style.scss

```diff
@@ -10,7 +10,10 @@
 			line-height: 1.6;
 		}
 
-		cite,
+		// The p + cite selector is for compatibility with older versions of the block that did not use
+		// footer. It has to be this specific in order to not target cite elements within the quoted
+		// paragraphs (or other quoted content if the block is updated to use nesting).
+		p + cite,
```

Comment:
Just for reference, the `text-align` property has no net effect on the citation on the front-end in `master`, and it still has no effect in this branch, since `text-align` has no apparent effect on inline elements. This back-compat selector is mainly just for the `font-size`, though that is relatively minor.

Since the font size could be considered unimportant, I would be happy to remove this back-compat selector if desired.